pam_rundir.c

```diff
@@ -404,8 +404,8 @@ pam_sm_open_session (pam_handle_t *pamh, int flags, int argc, const char **argv)
         secbits = prctl (PR_GET_SECUREBITS);
         if (secbits != -1)
             prctl (PR_SET_SECUREBITS, (unsigned long) secbits | SECBIT_NO_SETUID_FIXUP);
-        /* set euid so if we do create the dir, it is own by the user */
-        if (seteuid (pw->pw_uid) < 0)
+        /* set euid and egid so if we do create the dir, it is owned by the user */
+        if (seteuid (pw->pw_uid) < 0 || setegid (pw->pw_gid) < 0)
```

Comment:
pam_rundir creates `/run/user` owned by _root:root_ and mode _775_. If the target user is not member of group _root_, pam_rundir cannot create a directory inside it after switching uid/gid.

Reply:
Aha, no, it’s not an issue on branch `next`. The problem is that I backported this change to the latest release without 5ae2ee1a3411e973ebe56421850b7e08c23d6319. :/